### PR TITLE
Merges BundleXYZ into dev

### PR DESCRIPTION
**Pull request by jessemapel**
This is a large feature branch and has a lot of test data that needs to be checked in once this is merged. This also needs to be squashed down prior to merging so that it can be reverted in case of an emergency.

### CODE REVIEW

**Comment by jessemapel** [comment]
Closes #455

File: isis/src/control/apps/jigsaw/jigsaw.cpp [hunk 1]
```diff
@@ -166,26 +166,49 @@ BundleSettingsQsp bundleSettings(UserInterface &ui) {
...
+    coord3Sigma = ui.GetDouble("POINT_RADIUS_SIGMA");
+  }
+  if (ui.WasEntered("POINT_X_SIGMA")) {
+    coord1Sigma = ui.GetDouble("POINT_X_SIGMA");
+  }
+  if (ui.WasEntered("POINT_Y_SIGMA")) {
+    coord2Sigma = ui.GetDouble("POINT_Y_SIGMA");
+  }
+  if (ui.WasEntered("POINT_Z_SIGMA")) {
+    coord3Sigma = ui.GetDouble("POINT_Z_SIGMA");
```
Comment (SgStapleton) [inline isis/src/control/apps/jigsaw/jigsaw.cpp:201]: POINT_X/Y/Z_SIGMA will always overwrite POINT_LATITUDE/LONGITUDE/RADIUS_SIGMA values? Are these both allowed to be set in the gui?

File: isis/src/control/objs/BundleResults/BundleResults.cpp [hunk 2]
```diff
@@ -1450,13 +1428,43 @@ namespace Isis {
...
+
+  // *** TO DO *** Do we not need this method?
+  // SurfacePoint::CoordinateType BundleResults::coordTypeBundle() {
+  //   // Get the coordinate type from the output net if it exists.  Otherwise use the default.
+  //   SurfacePoint::CoordinateType type = SurfacePoint::Latitudinal;
+  //   if (m_outNet  && m_outNet->IsValid()) {
+  //       type = outputControlNet()->GetCoordType();
+  //   }
+  //   return type;
+  // }
```
Comment (SgStapleton) [inline isis/src/control/objs/BundleResults/BundleResults.cpp]: Was this meant to be removed, or are we leaving this in for the time being?

File: isis/src/control/objs/BundleAdjust/BundleAdjust.h [hunk 3]
```diff
@@ -141,7 +141,7 @@ namespace Isis {
    *                           updates to held images.
    *   @history 2011-06-27 Debbie A. Cook - and Ken Edmundson Added names to top header fields of
    *                           .csv output and fixed bugs in sparse output.
-   *   @history 2011-07-12 Ken Edmundson - Segmentation fault bugfix in OutputHeader method.
+   *   @history 2011-07-12 Ken Edmundson - Segmentation fault bugfix in OutputHeadsolveCholesker method.
```
Comment (SgStapleton) [inline isis/src/control/objs/BundleAdjust/BundleAdjust.h]: I imagine this was an accident as the method was never altered...?

File: isis/src/control/objs/BundleResults/unitTest.cpp [hunk 4]
```diff
@@ -346,6 +410,12 @@ int main(int argc, char *argv[]) {
       QString msg = "Unit Test failed. XML file [" + xmlPath + "not deleted.";
       throw IException(IException::Io, msg, _FILEINFO_);
     }
+
+    // deleted = qXmlFileR.remove();
+    // if (!deleted) {
+    //   QString msg = "Unit Test failed. XML file [" + xmlPath + "not deleted.";
+    //   throw IException(IException::Io, msg, _FILEINFO_);
+    // }
```
Comment (SgStapleton) [inline isis/src/control/objs/BundleResults/unitTest.cpp]: Does this need to stay?

File: isis/src/control/objs/BundleSolutionInfo/BundleSolutionInfo.h [hunk 5]
```diff
@@ -236,7 +249,7 @@ namespace Isis {
       QUuid              *m_id;
       QString             m_name;                        //!< Name of the bundle. Defaults to the id
       QString             m_runTime;                     //!< Run time of the bundle adjustment
-      FileName           *m_inputControlNetFileName;     //!< Input control network file name
+      FileName        *m_inputControlNetFileName;     //!< Input control network file name
```
Comment (SgStapleton) [inline isis/src/control/objs/BundleSolutionInfo/BundleSolutionInfo.h]:
```suggestion
      FileName            *m_inputControlNetFileName;     //!< Input control network file name
```

File: isis/src/control/objs/BundleUtilities/BundleControlPoint.cpp [hunk 6]
```diff
@@ -88,6 +100,9 @@ namespace Isis {
     m_adjustedSigmas = src.m_adjustedSigmas;
     m_weights = src.m_weights;
     m_nicVector = src.m_nicVector;
+    m_coordTypeReports = src.m_coordTypeReports;
+    m_coordTypeBundle = src.m_coordTypeBundle;
+    // *** TODO *** Why is the cholmodQMatrix not copied? Ask Ken
```
Comment (SgStapleton) [inline isis/src/control/objs/BundleUtilities/BundleControlPoint.cpp:105]: Was this ever decided? (May be irrelevant...?)

File: isis/src/control/objs/ControlPoint/ControlPoint.cpp [hunk 7]
```diff
@@ -973,14 +992,16 @@ namespace Isis {
...
+    // *** TODO *** This block appears to never get executed.  All cases fall into a
+    // previous if or else block.  It was not added in response to a bug according
+    // to history entries either.  I commented it out on 5/31/2017.  If no problems
+    // occur in the next 6 months, it can likely be deleted.
+    // else {
+    //   aprioriSurfacePoint.SetRectangular(
+    //     aprioriSurfacePoint.GetX(),
+    //     aprioriSurfacePoint.GetY(),
+    //     Displacement((zB / goodMeasures), Displacement::Kilometers));
+    // }
```
Comment (SgStapleton) [inline isis/src/control/objs/ControlPoint/ControlPoint.cpp]: Would this technically have been tested fully at this point?

File: isis/src/control/objs/ControlPoint/ControlPoint.h [hunk 8]
```diff
@@ -425,9 +428,12 @@ namespace Isis {
...
-        RadiusConstrained = 2,
+        Coord1Constrained = 0,
+        Coord2Constrained = 1,
+        Coord3Constrained = 2,
+//        LatitudeConstrained = 0,
+//        LongitudeConstrained = 1,
+//        RadiusConstrained = 2,
 //      XConstrained = 3,
 //      YConstrained = 4,
 //      ZConstrained = 5;
```
Comment (SgStapleton) [inline isis/src/control/objs/ControlPoint/ControlPoint.h]: Should these all be cleaned up?

File: isis/src/control/objs/ControlPoint/ControlPoint.h [hunk 9]
```diff
@@ -728,6 +735,9 @@ namespace Isis {
        * of the measures.
        */
       int numberOfRejectedMeasures;
+        
+      //! BundleControlPoint coordinate type.  This will default to Latitudinal, but is initialized from the parent Net.
+//      SurfacePoint::CoordinateType coordType;
```
Comment (SgStapleton) [inline isis/src/control/objs/ControlPoint/ControlPoint.h]: This not needed?

File: isis/src/qisis/objs/ConcurrentControlNetReader/ConcurrentControlNetReader.h [hunk 10]
```diff
@@ -52,7 +52,7 @@ namespace Isis {
    *
    * @internal
    *   @history 2017-08-09 Summer Stapleton - Added a try-catch block to handle invalid control
-   *                         networks. Fixes #5068.
+  *                         networks. Fixes #5068.
```
Comment (SgStapleton) [inline isis/src/qisis/objs/ConcurrentControlNetReader/ConcurrentControlNetReader.h]:
```suggestion
   *                         networks. Fixes #5068.
```

File: isis/src/qisis/objs/WindowTool/WindowTool.cpp [hunk 11]
```diff
@@ -187,7 +186,7 @@ namespace Isis {
       window->setGeometry(rect);
       window->move(position);
 
-      position.setX(position.x() + window->width());
+     position.setX(position.x() + window->width());
```
Comment (SgStapleton) [inline isis/src/qisis/objs/WindowTool/WindowTool.cpp]:
```suggestion
      position.setX(position.x() + window->width());
```

**Review by SgStapleton — CHANGES_REQUESTED** [review]

File: isis/src/control/apps/jigsaw/jigsaw.cpp (hunk 1 above)
Comment (jessemapel) [reply isis/src/control/apps/jigsaw/jigsaw.cpp:201]: The two sets of sigmas are mutually exclusive. If you look at `CONTROL_POINT_COORDINATE_TYPE_BUNDLE` it either excludes lat/lon/rad or x/y/z

**Review by SgStapleton — APPROVED** [review]

**Comment by SgStapleton** [comment]
Everything looks good!

**Pull request merged**